core/src/layers/prometheus_client.rs

```diff
@@ -77,68 +74,251 @@ use crate::*;
+    ///         )
+    ///         .finish();
+    ///     debug!("operator: {op:?}");
+    ///
+    ///     Ok(())
+    /// # }
+    /// ```
+    pub fn enable_path_label(mut self, level: usize) -> Self {
+        self.interceptor = self.interceptor.with_path_label_level(level);
+        self
     }
 }
 
 impl<A: Access> Layer<A> for PrometheusClientLayer {
     type LayeredAccess = observe::MetricsAccessor<A, PrometheusClientInterceptor>;
 
     fn layer(&self, inner: A) -> Self::LayeredAccess {
-        self.0.layer(inner)
+        observe::MetricsLayer::new(self.interceptor.clone()).layer(inner)
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Clone, Debug)]
 pub struct PrometheusClientInterceptor {
-    operation_duration_seconds: Family<OperationLabels, Histogram>,
-    operation_bytes: Family<OperationLabels, Histogram>,
-    operation_errors_total: Family<OperationErrorsTotalLabels, Counter>,
+    operation_duration_seconds: Family<OperationLabels, Histogram, HistogramConstructor>,
+    operation_bytes: Family<OperationLabels, Histogram, HistogramConstructor>,
+    operation_errors_total: Family<OperationLabels, Counter>,
+    path_label_level: usize,
+}
+
+#[derive(Clone)]
+struct HistogramConstructor {
+    buckets: Vec<f64>,
+}
+
+impl MetricConstructor<Histogram> for HistogramConstructor {
+    fn new_metric(&self) -> Histogram {
+        Histogram::new(self.buckets.iter().cloned())
+    }
+}
+
+impl Default for PrometheusClientInterceptor {
+    fn default() -> Self {
+        let operation_duration_seconds =
+            Family::<OperationLabels, _, _>::new_with_constructor(HistogramConstructor {
+                buckets: histogram::exponential_buckets(0.01, 2.0, 16).collect(),
+            });
+        let operation_bytes =
+            Family::<OperationLabels, _, _>::new_with_constructor(HistogramConstructor {
+                buckets: histogram::exponential_buckets(1.0, 2.0, 16).collect(),
+            });
+        let operation_errors_total = Family::<OperationLabels, Counter>::default();
+        Self {
+            operation_duration_seconds,
+            operation_bytes,
+            operation_errors_total,
+            path_label_level: 0,
+        }
+    }
+}
+
+impl PrometheusClientInterceptor {
+    fn with_operation_duration_seconds(mut self, buckets: Vec<f64>) -> Self {
+        self.operation_duration_seconds =
+            Family::<OperationLabels, _, _>::new_with_constructor(HistogramConstructor { buckets });
+        self
+    }
+
+    fn with_operation_bytes(mut self, buckets: Vec<f64>) -> Self {
+        self.operation_bytes =
+            Family::<OperationLabels, _, _>::new_with_constructor(HistogramConstructor { buckets });
+        self
+    }
+
+    fn with_path_label_level(mut self, level: usize) -> Self {
+        self.path_label_level = level;
+        self
+    }
+
+    fn register(&self, registry: &mut Registry) {
+        registry.register_with_unit(
+            "operation_duration",
```

Comment:
we can't use `observe::METRIC_OPERATION_DURATION_SECONDS.name()`, prometheus-client will generate `operation_duration_bytes_bytes` metric